cardano-chain-gen/test/Test/Cardano/Db/Mock/Unit/Conway/Governance.hs

```diff
@@ -73,72 +74,231 @@ drepDistr =
+    -- Wait for it to sync
+    assertBlockNoBackoff dbSync (length $ epoch1 <> epoch3)
+    -- Should now have a committee member
+    assertEqQuery
+      dbSync
+      Query.queryGovActionCounts
+      (1, 1, 0, 0)
+      "Unexpected governance action counts"
+  where
+    testLabel = "conwayNewCommittee"
 
-    -- It takes 2 epochs to enact a proposal--ratification will happen on the next
-    -- epoch and enacted on the following.
-    epochs <- Api.fillEpochs interpreter server 2
+rollbackNewCommittee :: IOManager -> [(Text, Text)] -> Assertion
+rollbackNewCommittee =
+  withFullConfig conwayConfigDir testLabel $ \interpreter server dbSync -> do
+    startDBSync dbSync
 
+    -- Register SPOs, DReps, and committee to vote
+    epoch1 <- initGovernance interpreter server
+    -- Propose, ratify, and enact a new committee member
+    epoch3 <- enactNewCommittee interpreter server
     -- Wait for it to sync
-    assertBlockNoBackoff dbSync (length epochs + 3)
+    assertBlockNoBackoff dbSync (length $ epoch1 <> epoch3)
     -- Should now have a committee member
     assertEqQuery
       dbSync
       Query.queryGovActionCounts
       (1, 1, 0, 0)
       "Unexpected governance action counts"
+
+    -- Rollback the last 2 blocks
+    epoch1' <- rollbackBlocks interpreter server 2 epoch3
+    -- Wait for it to sync
+    assertBlockNoBackoff dbSync (length $ epoch1 <> epoch1')
+    -- Should not have a new committee member
+    assertEqQuery
+      dbSync
+      Query.queryGovActionCounts
+      (1, 0, 0, 0)
+      "Unexpected governance action counts"
+
+    -- Fast forward to next epoch
+    epoch2' <- Api.fillUntilNextEpoch interpreter server
+    assertBlockNoBackoff dbSync (length $ epoch1 <> epoch1' <> epoch2')
+    -- Should now have 2 identical committees
+    assertEqQuery
+      dbSync
+      (Query.queryEpochStateCount 3)
+      2
+      "Unexpected epoch state count for epoch 3"
   where
-    testLabel = "conwayNewCommittee"
+    testLabel = "conwayRollbackNewCommittee"
 
-updateConstitution :: IOManager -> [(Text, Text)] -> Assertion
-updateConstitution =
+chainedNewCommittee :: IOManager -> [(Text, Text)] -> Assertion
+chainedNewCommittee =
   withFullConfig conwayConfigDir testLabel $ \interpreter server dbSync -> do
     startDBSync dbSync
 
-    -- Add stake
-    void (Api.registerAllStakeCreds interpreter server)
+    -- Register SPOs, DReps, and committee to vote
+    epoch1 <- initGovernance interpreter server
+
+    -- Wait for it to sync
+    assertBlockNoBackoff dbSync (length epoch1)
+    -- Should start with 4 committee members
+    assertEqQuery
+      dbSync
+      (Query.queryCommitteeMemberCountByTxHash Nothing)
+      4
+      "Unexpected committee member count"
 
-    -- Register a DRep and delegate votes to it
-    void (Api.registerDRepsAndDelegateVotes interpreter server)
+    let
+      -- Propose a new committee member
+      committee1Hash = "e0a714319812c3f773ba04ec5d6b3ffcd5aad85006805b047b082541"
```

Comment:
very minor but is it worth abstracting this hash out as it's used in 2 places? Happy to leave as is tbh.